### PR TITLE
feat: handle offline screen

### LINKED ISSUE #2249: Handle offline-first screen
Tell users they are out of the internet and once they are back, they should see NFTs

image

### PR DESCRIPTION
**Pull request by KngZhi**
Refs: #2249

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2249 
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer?target=HgpGWWmqrw8t6viwSy6LPcTVr2SuVPKWW5ArdNp2xCJY2U4&usdamount=100&donation=true)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

[Watch the video](https://vimeo.com/691279398/57ca12d273)

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Hey! 
You're handling offline only on Gallery view, there could be multiple solution to fix this problem. you could try to:
- create a middleware to redirect to an error page when the internet connection of a user is not present
- move the network verification on layout level

By the way Nuxt offers a [connection checkers](https://nuxtjs.org/docs/internals-glossary/$nuxt/#connection-checker) which can certainly help you

**Comment by KngZhi** [comment]
> Hey! You're handling offline only on Gallery view, there could be multiple solution to fix this problem. you could try to:
> 
> * create a middleware to redirect to an error page when the internet connection of a user is not present
> * move the network verification on layout level
> 
> By the way Nuxt offers a [connection checkers](https://nuxtjs.org/docs/internals-glossary/$nuxt/#connection-checker) which can certainly help you

Okay, I get it. I would reimplement.

**Comment by yangwao** [comment]
> By the way Nuxt offers a [connection checkers](https://nuxtjs.org/docs/internals-glossary/$nuxt/#connection-checker) which can certainly help you

seems we could use it for #2512

File: layouts/full-width-layout.vue [hunk 1]
```diff
@@ -2,6 +2,11 @@
   <div class="min-h-full is-flex is-flex-direction-column">
     <Navbar />
     <main class="is-flex-grow-1">
+      <Error
+        v-if="$nuxt.isOffline"
+        :hasImg="false"
+        errorTitle="Offline Detected"
+        errorSubtitle="Please check your network connections" />
       <Nuxt />
```
Comment (roiLeo) [inline layouts/full-width-layout.vue:10]:
```suggestion
      <Nuxt v-else />
```

**Review by roiLeo — APPROVED** [review]
lgtm

**Comment by yangwao** [comment]
I guess it's fair.
Now would be amazing to make handle offline-first v2 to add things 
- like notice user visually that's it's offline, but keep showing content - i.e. now when I'm offline and have cached pictures, why I can't see them if I don't interact with them?
- disable buttons probably 
- write service worker to cache stuff, i.e. add the ability for user return to pages were previously been?
- learn from https://offlinefirst.org/
- https://hasura.io/blog/design-guide-to-offline-first-apps/
- https://techbeacon.com/app-dev-testing/offline-first-web-mobile-apps-top-frameworks-components

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
for some reason, payout bot can not read your address, can you @KngZhi double check?

**Comment by KngZhi** [comment]
@yangwao I have changed the location.

Besides, I want to mention that [Rewards](https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#sharing-your-address-to-receive-the-reward) section on CONTRIBUTING is kind confusing

> Entering an amount like 100 USD and clicking the button **Copy Payout address** will copy the link to your clipboard

`Copy Payout address` should change to `COPY PAYMENT LINK`

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
> @yangwao I have changed the location.
> 
> Besides, I want to mention that [Rewards](https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md?rgh-link-date=2022-03-31T00%3A26%3A17Z#sharing-your-address-to-receive-the-reward) section on CONTRIBUTING is kind confusing
> 
> > Entering an amount like 100 USD and clicking the button **Copy Payout address** will copy the link to your clipboard
> 
> `Copy Payout address` should change to `COPY PAYMENT LINK` #2717
> 
> Since I've been tested for various addresses if using `COPY PAYOUT ADDRESS` the copied link remains the same.

yes, it's changing quite fast and docs we're obsolete, thank you for updating it!

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
> Missing address to be paid at 2645

strange, I'll process it manually and check later 

https://kodadot.xyz/transfer?target=DaWoVNNhBTdqJ5iPN3UcaLeamkPpJeGWdrs8LhvX9TQbadc&usdamount=100&donation=true

**Comment by yangwao** [comment]
payout sent https://kusama.subscan.io/extrinsic/0xd83215fd8de17e2e1bf0e3264413888e841e844e6040f791993bd36d79faead3

**Pull request merged**